### PR TITLE
Enable support for Busybox tools.

### PR DESCRIPTION
**Pull request by kacfengine**
When running on Linux, CFEngine core assumes that the GNU utilities are
used.  Howver on embedded devices we commonly (always?) use busybox.
Thus 'out-of-the-box' certain functions do not work.  For example, it's
not possible to bootstrap a device because busybox /bin/ps accepts
different arguments.  The patch requires __BUSYBOX__ to be defined in
the build environment for busybox targets - hopefully there is a better
way where it can be detected automatically at runtime.

### CODE REVIEW

**Comment by kacfengine** [comment]
trigger build

File: libpromises/systype.c [hunk 1]
```diff
@@ -95,6 +97,7 @@ const char *const VPSOPTS[] =
     [PLATFORM_CONTEXT_AIX] =  "-N -eo user,pid,ppid,pgid,pcpu,pmem,vsz,ni,stat,st=STIME,time,args",       /* aix */
     /* Note: keep in sync with GetProcessOptions()'s hack for Linux 2.4 */
     [PLATFORM_CONTEXT_LINUX] = "-eo user,pid,ppid,pgid,pcpu,pmem,vsz,ni,rss:9,nlwp,stime,etime,time,args",/* linux */
+    [PLATFORM_CONTEXT_BUSYBOX] = "",                  /* linux / busybox */
```
Comment (jimis) [inline libpromises/systype.c]: If I recall, it just needs the "w" argument to avoid truncation. I'll login to our router to find out. :-p

**Comment by jimis** [comment]
+1

**Comment by kacfengine** [comment]
trigger build

File: libpromises/systype.c (hunk 1 above)
Comment (jimis) [inline libpromises/systype.c]: Indeed, "w" seems to be the only acceptable parameter, at least for recent versions of busybox. I'll directly commit that change, OK @kacfengine?

Comment (kacfengine) [inline libpromises/systype.c]: Ok with me!

**Pull request merged**